Каперы: Новый Горизонт

Maggotson opened the pull request:
Я, Хэм Мэгготсон, презентую вам свою долгожданную разработку "Каперы: Новый горизонт".

## О ПР`е
<!-- Что вы поменяли? -->
 Он добавляет большое количество контента, но самая важная часть его - переработка режима капера. 
 
 **ДИСКЛЕЙМЕР СРОЧНО!!!!!** Подразумевается, чтобы каперы не запускались на публичной сборке, так как там отсутствует намапленный шаттл. Подразумевается, что на приватной сборке будет добавлен шаттл (Намапплю я его на стейдже, там надо добавить прототипы из этой разработки, а также из приватного репозитория, где присутствуют прототипы связанные с хирургией), а также там будут включен режим каперов в голосовании, а также каперы будут добавлены в секрет и ВВО. 
 
 
 Режим полностью переписан с нуля, дополнен функционалом, таким как: перенос персонализации на энтити капера, условие победы (да!!!), а также новый аплинк (называющийся Фрибутером), позволяющий покупать за дублоны различные предметы: оружие, шмотки, пиво, и многое другое. "Стопе, что за дублоны, матрос?" - можете спросить вы. Дублоны - аналог кредитов, которые можно получить, продав их использую пиратскую консоль продажи. Чем она отличается от обычной консоли продаж? А тем, что вы можете продавать ВСЕ!!! ДА!!! Ядерные бомбы, людей, никаких ограничений! Даже если вы захотите продать хай-риски, то не спешите, их цена слишком мала. Такой расклад не устраивает каперские будни, поэтому теперь каперы могут многократно увеличивать цену ЛЮБЫХ хайрисков. Но как? Каперская разведка установила, что в хайрисках есть GPS трекеры NT!!! Из-за них хай-риски и стоят настолько мало. Исправляем ситуацию, используя извлекатель gps трекеров 2000. Имеется на каждом судне пиратском!! К слову, бабки зарабатывать каперам однозначно надо, ведь чтобы им победить надо заработать (вроде как) 300к (меняется cvar-ами, оу да!!!). Но как зарабатывать капусту? Есть много вариантов. Мы уже упомянули продажу хайрисков, но есть и другие варианты. Первый вариант - высасывать штормовую мощь проводов космических суден (почти как майнер). Устанавливается штормовая пучина на ВВ провода, а потом при полном заряде имеет великолепную цену!!! Это крутец!!! Также можно устроить пивоварню. Сварив эссенцию алкоголя вы сможете получить невероятное количество денег, либо выпить этот божественный напиток для получения различных баффов и хила! Насчет баффов - добавлен новый эффект реагента, увеличивающий резисты игрока. Крутец !!!. Как варится эссенция алкоголя? Из эссенций пива, рома, вина, а они в свою очередь из других прекрасных напитков. Нужно кучу ресурсов, чтобы сварить это, но это стоит того. Что я могу сказать. Каперы настолько крутые,что поменяли реальность. Как? Теперь пайп бомбу НУЖНО поджигать горячими предметами! (К сожалению теперь ручками не поджечь трубчатую бомбу!). К слову, благодаря крутецкой работе Макс Габа у каперов есть более крутые шмотки!! Очень красивые. Их можно будет получить с лутбокса, где также падают бафающие пояса. Спасибо макс габ, очень круто. К слову есть полный перевод на англ, не знаю зачем, но я постарался. Я очень горжусь своей разработкой, на неё я потратил большое количество времени. 
## Технические детали
<!-- Сводка изменений кода для более удобного просмотра. -->
Добавление систем, связанных с GPS трекерами, позволяющие увеличить цены хай-рисков
Добавление систем, связанных с новым эффектом реагентов, позволяющим изменять резисты
Добавление систем, связанных с переработанным режимом - Каперами.
(!!ПОДРОБНЕЕ ВЫШЕ!!)
## Изменения кода официальных разработчиков
<!-- Сводка изменений кода визардов для более удобного просмотра (Если есть). -->
Небольшие дополнения в системе тригерров бомб, систем карго паллет, системы определния цены и PowerSink для добавления нового функционала. (Параметры взрыва в PowerSink, система тригерров: изменение Update, итд, паллеты: добавление функционала обхода blacklist, система определения цены: добавлен учет цены без GPS трекера)(!!ПОДРОБНЕЕ ВЫШЕ!!)

## Благодарность
Макс Габ - Спасибо за крутецкую одежду
spacerangeer - Спасибо за крутецкие сочные спрайты
Phoenix - Спасибо за крутецкий спрайт
Обернштадофисер Бобр - За поддержку и участие в разработке каперов.
Девы Imperial Space - За поддержку в трудный момент
krevetkqqq - За поддержку и участие в разработке каперов

А также все сообщество Imperial Space
![sea-shanties-dancing-pirate](https://github.com/user-attachments/assets/2bcf1d8f-dcb2-44cc-b0ca-80f64e5c743d)

Maggotson committed:
Каперы: Новый Горизонт

Я, Хэм Мэгготсон, презентую вам свою долгожданную разработку "Каперы: Новый горизонт". Она добавляет большое количество контента, но самая важная часть его - переработка режима капера. Полностью переписан с нуля, дополнен функционалом, таким как: перенос персонализации на энтити капера, условие победы (да!!!), а также новый аплинк (называющийся Фрибутером), позволяющий покупать за дублоны различные предметы: оружие, шмотки, пиво, и многое другое. "Стопе, что за дублоны, матрос?" - можете спросить вы. Дублоны - аналог кредитов, которые можно получить, продав их использую пиратскую консоль продажи. Чем она отличается от обычной консоли продаж? А тем, что вы можете продавать ВСЕ!!! ДА!!! Ядерные бомбы, людей, никаких ограничений! Даже если вы захотите продать хай-риски, то не спешите, их цена слишком мала. Такой расклад не устраивает каперские будни, поэтому теперь каперы могут многократно увеличивать цену ЛЮБЫХ хайрисков. Но как? Каперская разведка установила, что в хайрисках есть GPS трекеры NT!!! Из-за них хай-риски и стоят настолько мало. Исправляем ситуацию, используя извлекатель gps трекеров 2000. Имеется на каждом шаттле NT!!! К слову, бабки зарабатывать каперам однозначно надо, ведь чтобы им победить надо заработать (вроде как) 300к (меняется cvar-ами, оу да!!!). Но как зарабатывать капусту? Есть много вариантов. Мы уже упомянули продажу хайрисков, но есть и другие варианты. Первый вариант - высасывать штормовую мощь проводов космических суден (почти как майнер). Устанавливается штормовая пучина на ВВ провода, а потом при полном заряде имеет великолепную цену!!! Это крутец!!! Также можно устроить пивоварню. Сварив эссенцию алкоголя вы сможете получить невероятное количество денег, либо выпить этот божественный напиток для получения различных баффов и хила! Насчет баффов - добавлен новый эффект реагента, увеличивающий резисты игрока. Крутец !!!. Как варится эссенция алкоголя? Из эссенций пива, рома, вина, а они в свою очередь из других прекрасных напитков. Нужно кучу ресурсов, чтобы сварить это, но это стоит того. Что я могу сказать. Каперы настолько крутые,что поменяли реальность. Как? Теперь пайп бомбу НУЖНО поджигать горячими предметами! (К сожалению теперь ручками не поджечь трубчатую бомбу!). К слову, благодаря крутецкой работе Макс Габа у каперов есть более крутые шмотки!! Очень красивые. Их можно будет получить с лутбокса, где также падают бафающие пояса. Спасибо макс габ, очень круто. К слову есть полный перевод на англ, не знаю зачем, но я постарался. Я очень горжусь своей разработкой, на неё я потратил большое количество времени. ДИСКЛЕЙМЕР СРОЧНО!!!!! Подразумевается, что каперы не запускаются на публичной сборке, так как там отсутствует намапленный шаттл. Подразумевается, что на приватной сборке будет добавлен намапленный шаттл (Намапплю я его на стейдже, там надо добавить прототипы из этой разработки), а также там будут включен режим каперов, а также каперы будут добавлены в секрет и вво. Спасибо за внимание, гайс, всех люблю.

krevetkqqq commented:
из ветки develop. переделывай

JustGostuY reviewed (changes requested):
Все новые файлы должны быть по пути .../Imperial/...
Свести к минимуму изменения оригинального кода - это значит, что максимум допускается вызов необходимой функции в оригинальном коде, обозначенный комментарием "Imperial *Название разработки*"

Maggotson commented:
> Все новые файлы должны быть по пути .../Imperial/... Свести к минимуму изменения оригинального кода - это значит, что максимум допускается вызов необходимой функции в оригинальном коде, обозначенный комментарием "Imperial _Название разработки_"

Новые файлы, а если новая ID карта, то мне тоже их по /Imperial/ сувать или уже к существующему yml всунуть?

JustGostuY commented:
> > Все новые файлы должны быть по пути .../Imperial/... Свести к минимуму изменения оригинального кода - это значит, что максимум допускается вызов необходимой функции в оригинальном коде, обозначенный комментарием "Imperial _Название разработки_"
> 
> Новые файлы, а если новая ID карта, то мне тоже их по /Imperial/ сувать или уже к существующему yml всунуть?

/Resources/Prototypes/Imperial/*Ваша разработка*/*Файлы с новым контентом*

Maggotson committed:
Pirates: New Horizon Mini Fix

Movin some files

Molokvasik commented:
Неужели это легендарный камбек каперов!??

Maggotson committed:
Pirates: New Horizon Mini Fix (Again)

Removed unnecessary files

Maggotson commented:
> Все новые файлы должны быть по пути .../Imperial/... Свести к минимуму изменения оригинального кода - это значит, что максимум допускается вызов необходимой функции в оригинальном коде, обозначенный комментарием "Imperial _Название разработки_"

Cделал некоторые изменения, что делать насчет кода визардов пока даже не знаю.

Qsan163 commented:
Требую добавить эту мелодию в музыкальный автомат в баре в честь этого обновления :
https://youtu.be/Qy6MrsvKyqU

Qsan163 commented:
> > Все новые файлы должны быть по пути .../Imperial/... Свести к минимуму изменения оригинального кода - это значит, что максимум допускается вызов необходимой функции в оригинальном коде, обозначенный комментарием "Imperial _Название разработки_"
> 
> Cделал некоторые изменения, что делать насчет кода визардов пока даже не знаю.

можешь сделать свой компонент сделав ctrl+c и ctrl+v , и добавив в названии Imperial, а старые системы вернуть на место

Maggotson committed:
Merge branch 'imperial-space:develop' into develop

Maggotson committed:
Quick Fix

Fixed errors

Maggotson committed:
Pirates: New Horizon; Code Replacement

Placed 90% of code in corresponding folders + comments added

Maggotson committed:
Pirates: New Horizon; Trigger System Fix

yeah

Maggotson committed:
Prototype Pirates: New Horizon Quick Fix

Maggotson committed:
Merge branch 'imperial-space:develop' into develop

Maggotson committed:
Prototype Comment Add Pirates: New Horizon

Maggotson committed:
Merge branch 'develop' of https://github.com/Maggotson/SS14-public into develop

JustGostuY reviewed (changes requested):
`Content.Server/Roles/PirateRoleComponent.cs`
`Content.Shared/Cargo/Events/GPSTrackerRemoveDoAfterEvent.cs`
`Content.Shared/Chemistry/Components/ResistModifierMetabolismComponent.cs`
`Resources/Audio/Effects/weak_bang.ogg`
`Resources/Locale/en-US/guidebook/chemistry/resistancechange.ftl`
Убрать файлы из папок визов

Вижу, что некоторые прототипы можно не изменять в файлах визов, а перенести в свои файлы.

JustGostuY commented on Resources/Prototypes/Entities/Objects/Tools/t-ray.yml:
```diff
@@ -25,3 +25,7 @@
+  #Imperial Space Pirates: New Horizon
+  - type: Tag
+    tags:
+    - TrayScanner
```
Зачем?

Maggotson replied:
![image](https://github.com/user-attachments/assets/10e677df-6858-493d-92a3-60ff2b8c3510)

krevetkqqq replied:
```suggestion
    - TrayScanner
    #Imperial Space Pirates: New Horizon end
```

JustGostuY commented on Resources/Prototypes/Entities/Objects/Weapons/Bombs/pipebomb.yml:
```diff
@@ -10,11 +10,6 @@
     - state: base
       map: ["enum.TriggerVisualLayers.Base"]
     - state: wires
-  - type: OnUseTimerTrigger # todo: make it activate through welder/lighter/fire instead
```
Зачем?

Maggotson replied:
**"todo: make it activate through welder/lighter/fire instead"**

JustGostuY replied:
> **"todo: make it activate through welder/lighter/fire instead"**

все изменения в прототипе для чего? Создайте такой же, но свой

Maggotson replied:
> > **"todo: make it activate through welder/lighter/fire instead"**
> 
> все изменения в прототипе для чего? Создайте такой же, но свой

Мне надо просто пайпбомбу поменять, если отдельный прот делать, то тогда надо будет визардский крафт поменять ☔

JustGostuY commented on Resources/Prototypes/Entities/Objects/Weapons/Melee/e_sword.yml:
```diff
@@ -250,8 +250,9 @@
         types:
-            Slash: 10
-            Heat: 12
+        #Imperial Space Pirates: New Horizon
```
Зачем?

Maggotson replied:
бафф

JustGostuY replied:
> бафф

Не требуется

Maggotson replied:
> > бафф
> 
> Не требуется

Ладно

JustGostuY commented on Resources/Textures/Clothing/OuterClothing/Coats/pirate.rsi/meta.json:
```diff
@@ -1,30 +1,30 @@
-  },
-  "states": [
-    {
-      "name": "icon"
```
Не изменять, добавить зареспрайченый объект в свои файлы

JustGostuY commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/pirate.rsi/meta.json:
```diff
@@ -1,30 +1,29 @@
-      "name": "icon"
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "made by max_gab",
```
Не изменять, добавить зареспрайченый объект в свои файлы

JustGostuY commented on Resources/Textures/Structures/Machines/computers.rsi/meta.json:
```diff
@@ -1718,6 +1718,28 @@
+        },
+        {
+            "name": "pirate",
+            "directions": 4,
```
Нет комментария. Также пометить `Resources/Textures/Structures/Machines/computers.rsi/pirate.png` - "Imperial"

The-Fiafi replied:
Пожалуйста, уберите свой спрайт из папок визов. Сделайте как вам указал разработчик выше. На  саму консоль накиньте компонент спрайта с измененными путями

The-Fiafi replied:
Комментарии ставить не надо. Тем более это запрещено конвенцией

Maggotson replied:
> Комментарии ставить не надо. Тем более это запрещено конвенцией

Что за конвенция?

JustGostuY commented on Resources/Prototypes/Entities/Objects/Misc/identification_cards.yml:
```diff
@@ -691,8 +697,8 @@
-    - NuclearOperative
-    - SyndicateAgent
+    #Imperial Space Pirates: New Horizon
+    - Pirate
```
Сделать свою ID карту, в соответствующей разработке папке

JustGostuY reviewed (changes requested):
Более ответственно отнеситесь к требованиям :

> Свести к минимуму изменения оригинального кода

The-Fiafi reviewed (changes requested):
Все строчки комментируются по схеме `Imperial Space <Название вашей разработки>`.

Если подряд идущих строчек от 2 до 3, то вы ставите комментарии перед началом ваших изменений `Imperial Space <Название вашей разработки> Start` и заканчиваете строчкой `Imperial Space <Название вашей разработки> End`

Maggotson committed:
Merge branch 'develop' into develop

Maggotson committed:
Pirates: New Horizon; Quick Fix

Maggotson commented:
> Все строчки комментируются по схеме `Imperial Space <Название вашей разработки>`.
> 
> Если подряд идущих строчек от 2 до 3, то вы ставите комментарии перед началом ваших изменений `Imperial Space <Название вашей разработки> Start` и заканчиваете строчкой `Imperial Space <Название вашей разработки> End`

Я все сделал?

Maggotson committed:
Pirates New Horizon Quick Fix

Maggotson commented:
Ау?

Maggotson commented:
@The-Fiafi я все сделал?

Stevensegege commented:
Не пингуй ты их, как будет время и посчитают нужным посмотрят. А от этого быстрей не рассмотрят.

Maggotson commented:
Я жду.

Poeeno commented:
Разработчики обязательно разберутся со всеми вопросами. Сейчас у них нет времени быстро всё проверить. Не волнуйтесь, они обязательно это сделают.

JustGostuY commented on Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:
```diff
@@ -245,7 +245,7 @@ private void GetPalletGoods(EntityUid gridUid, out HashSet<EntityUid> toSell, ou
         toSell = new HashSet<EntityUid>();
 
-        foreach (var (palletUid, _, _) in GetCargoPallets(gridUid, BuySellType.Sell))
+        foreach (var (palletUid, palletComp, _) in GetCargoPallets(gridUid, BuySellType.Sell))
```
Нет комментария

JustGostuY commented on Content.Server/Cargo/Components/CargoPalletComponent.cs:
```diff
@@ -23,4 +23,12 @@ public sealed partial class CargoPalletComponent : Component
+    /// If pallets allows selling blacklisted entities (e.g. players, nukes)
+    /// </summary>
+    [DataField]
+    public bool AvoidSellBlacklist = false;
```
Подробно объяснить - "зачем это нужно?"

krevetkqqq replied:
@JustGostuY `/// If pallets allows selling blacklisted entities (e.g. players, nukes)
    /// </summary>`

krevetkqqq replied:
@Maggotson сделай отдельный компонент и на ивент который там подписывайся и `ev.Cancelled = true;`

JustGostuY commented on Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:
```diff
@@ -267,7 +267,9 @@ private void GetPalletGoods(EntityUid gridUid, out HashSet<EntityUid> toSell, ou
-                    continue;
+                    //Imperial Space Pirates: New Horizon
+                    if(palletComp.AvoidSellBlacklist == false)
+                        continue;
```
Подробно объяснить - зачем это нужно?

JustGostuY commented on Content.Server/Cargo/Systems/PricingSystem.cs:
```diff
@@ -16,6 +16,7 @@
 using Robust.Shared.Prototypes;
 using Robust.Shared.Utility;
 using System.Linq;
+using Content.Server.Imperial.PiratesNewHorizon.GPS.Components;
```
Нет комментария

JustGostuY commented on Content.Server/Explosion/EntitySystems/TriggerSystem.OnUse.cs:
```diff
@@ -110,6 +111,10 @@ private void OnRandomTimerTriggerMapInit(Entity<RandomTimerTriggerComponent> ent
+        // Imperial Space Pirates: New Horizon
+        if(!TryComp<TriggerOnHeatComponent>(ent, out var heatTriggerComp))
+            return;
+        _triggerHeat.OnRandomTimerHeatTriggerMapInit(ent, ref args);
```
Закомментировать через start-end. Показать на примере работу новых добавленных механик - добавить видеообзор в шапку ПР`а, в раздел "Медиа"

JustGostuY commented on Content.Server/Explosion/EntitySystems/TriggerSystem.OnUse.cs:
```diff
@@ -5,13 +5,14 @@
 {
     [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
-
+    [Dependency] private readonly TriggerSystemOnHeat _triggerHeat = default!;
```
Нет комментария

JustGostuY commented on Content.Server/Explosion/EntitySystems/TriggerSystem.OnUse.cs:
```diff
@@ -5,13 +5,14 @@
 using Content.Shared.Verbs;
-
+using Content.Server.Imperial.PiratesNewHorizon.Trigger.Systems;
+using Content.Server.Imperial.PiratesNewHorizon.Trigger.Components;
```
Нет комментария

JustGostuY commented on Content.Server/PowerSink/PowerSinkComponent.cs:
```diff
@@ -41,5 +41,22 @@ public sealed partial class PowerSinkComponent : Component
+        /// </summary>
+        [DataField("explodeOnFullCharge")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public bool ExplodeOnFullCharge = true;
```
Зачем и почему по умолчанию `true`?

JustGostuY commented on Resources/Audio/Effects/attributions.yml:
```diff
@@ -240,4 +240,4 @@
   copyright: 'created by Vrymaa on Freesound and modified by Chaboricks'
   license: "CC0-1.0"
-  source: https://freesound.org/people/Vrymaa/sounds/785128/
+  source: https://freesound.org/people/Vrymaa/sounds/785128/
```
Лишнее изменение файла - откатить

JustGostuY commented on Resources/Locale/ru-RU/guidebook/chemistry/effects.ftl:
```diff
@@ -380,3 +380,16 @@ reagent-effect-guidebook-add-to-solution-reaction =
         [1] Заставляет
        *[other] заставляют
     } химикаты, применённые к объекту, добавиться во внутренний контейнер для растворов этого объекта
+reagent-effect-guidebook-resistance-change =
```
Нет комментария start-end

JustGostuY commented on Resources/Locale/ru-RU/prototypes/roles/antags.ftl:
```diff
@@ -42,3 +42,9 @@ roles-antag-syndicate-agent-sleeper-objective = Разновидность аг
+roles-antag-pirate-objective = Сгруппируйтесь со своей коммандой на вояж к разграбу станции!
+
+roles-antag-captainpirate-name = Капитан пират
+roles-antag-captainpirate-objective = Ахой! Укажи путь своему экипажу к богатству (путем разграба). Не подведи Муниципальность Горизонта!
```
Перенести изменения в свои файлы локализации

JustGostuY commented on Resources/Prototypes/Catalog/Fills/Crates/antag.yml:
```diff
@@ -29,3 +31,5 @@
         amount: 4
       - id: WeaponShotgunBlunderbuss
         amount: 2
+  - type: AccessReader
```
Нет комментария

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml:
```diff
@@ -253,6 +257,10 @@
+    #Imperial Space Pirates: New Horizon
+    - type: Tag
+      tags:
+      - HudChemical
```
Зачем?

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/glasses.yml:
```diff
@@ -65,6 +65,10 @@
+    #Imperial Space Pirates: New Horizon
+  - type: Tag 
+    tags:
+    - Meson
```
Зачем?

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -29,6 +29,10 @@
+    #Imperial Space Pirates: New Horizon
+  - type: Tag
+    tags:
+    - HudDiagnostic
```
Зачем?

krevetkqqq replied:
```suggestion
    - HudDiagnostic # Imperial Space Pirates: New Horizon End
```

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -84,6 +88,10 @@
+  #Imperial Space Pirates: New Horizon
+  - type: Tag
+    tags:
+    - HudBeer
```
Зачем?

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -235,6 +243,13 @@
+    - EyepatchMedical
+  - type: Construction
+    graph: MedicalEyepatchHud
+    node: medicalEyepatch
```
Зачем?

krevetkqqq replied:
```suggestion
    node: medicalEyepatch
    #Imperial Space Pirates: New Horizon end
```

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -251,6 +266,13 @@
+    - EyepatchSecurity
+  - type: Construction
+    graph: SecurityEyepatchHud
+    node: securityEyepatch
```
Зачем?

krevetkqqq replied:
```suggestion
    node: securityEyepatch
    #Imperial Space Pirates: New Horizon end
```

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -267,6 +289,14 @@
+  - type: Construction
+    graph: BeerEyepatchHud
+    node: beerEyepatch
+
```
Зачем?

krevetkqqq replied:
```suggestion
#Imperial Space Pirates: New Horizon end
```

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -283,6 +313,13 @@
+    - EyepatchDiagnostic
+  - type: Construction
+    graph: DiagnosticEyepatchHud
+    node: diagnosticEyepatch
```
Зачем?

krevetkqqq replied:
```suggestion
    node: diagnosticEyepatch
    #Imperial Space Pirates: New Horizon end
```

JustGostuY commented on Resources/Prototypes/Entities/Clothing/Eyes/misc.yml:
```diff
@@ -28,6 +28,10 @@
+    #Imperial Space Pirates: New Horizon
+  - type: Tag
+    tags:
+      - Eyepatch
```
Зачем?

krevetkqqq replied:
```suggestion
      - Eyepatch
      #Imperial Space Pirates: New Horizon end
```

JustGostuY commented on Resources/Prototypes/Entities/Mobs/NPCs/pets.yml:
```diff
@@ -36,6 +36,12 @@
   - type: StealTarget
     stealGroup: AnimalIan
+  - type: StaticPrice
+    price: 2000
```
Нет комментария

krevetkqqq replied:
есть

JustGostuY commented on Resources/Prototypes/Entities/Objects/Devices/hand_teleporter.yml:
```diff
@@ -14,6 +14,12 @@
   - type: StealTarget
     stealGroup: HandTeleporter
+  - type: StaticPrice
+    price: 2000
```
Нет комментария

JustGostuY commented on Resources/Prototypes/Entities/Objects/Misc/identification_cards.yml:
```diff
@@ -119,8 +119,14 @@
     - WhitelistChameleonIdCard
     - HighRiskItem
+  - type: StaticPrice
+    price: 2000
```
Нет комментария

JustGostuY commented on Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:
```diff
@@ -159,6 +159,12 @@
   - type: StealTarget
     stealGroup: JetpackCaptainFilled
+  - type: StaticPrice
+    price: 2000
```
Нет комментария

JustGostuY commented on Resources/Prototypes/Entities/Objects/Weapons/Bombs/pipebomb.yml:
```diff
@@ -30,6 +25,17 @@
+    visuals:
+      enum.Trigger.TriggerVisuals.VisualState:
+        enum.ConstructionVisuals.Layer:
+          Primed: { state: primed }
```
Новые строки в комментарий формата start-end

JustGostuY commented on Resources/Prototypes/Imperial/PiratesNewHorizon/Misc/gamerules.yml:
```diff
@@ -0,0 +1,96 @@
+      fallbackRoles: [ CaptainPirate ]
+      spawnerPrototype: SpawnPointPrivateer
+      max: 14
+      playerRatio: 1
```
Что за роль? Почему макс 14 ролей на 20 человек с playerRatio 1?

Maggotson replied:
Опечатка. Исправлено

Maggotson replied:
Со следующим комитом будет

JustGostuY reviewed (changes requested):
Новые или изменённые мапы, гриды создать/скопировать в Imperial/Maps/название разработки.
Цены с gps трекером снизить в 1.5/2 раза
Добавить медиа контент в раздел ПР`а "Медиа" - видеоролик с обзором Ваших антагонистов и изменений визовского кода

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -457,6 +457,10 @@
     price: 750
   - type: StealTarget
     stealGroup: ClothingOuterHardsuitRd
+    #Imperial Space Pirates: New Horizon
```
Комментируйте каждую строчку. Либо закрывайте ваши изменения в комментариях `Imperial Space название_разработки Start/End`

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml:
```diff
@@ -61,6 +61,10 @@
     price: 750
   - type: StealTarget
     stealGroup: ClothingShoesBootsMagAdv
+  #Imperial Space Pirates: New Horizon
```
Комментируйте каждую строчку. Либо закрывайте ваши изменения в комментариях `Imperial Space название_разработки Start/End`

The-Fiafi commented on Resources/Prototypes/Entities/Mobs/NPCs/pets.yml:
```diff
@@ -36,6 +36,12 @@
     stealGroup: AnimalIan
+  - type: StaticPrice
+    price: 2000
+    #Imperial Space Pirates: New Horizon
```
Комментируйте каждую строчку. Либо закрывайте ваши изменения в комментариях `Imperial Space название_разработки Start/End`

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Devices/Circuitboards/computer.yml:
```diff
@@ -253,7 +253,10 @@
       tags:
       - HighRiskItem
-
+    #Imperial Space Pirates: New Horizon
```
Комментируйте каждую строчку. Либо закрывайте ваши изменения в комментариях `Imperial Space название_разработки Start/End`

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Devices/hand_teleporter.yml:
```diff
@@ -14,6 +14,12 @@
     stealGroup: HandTeleporter
+  - type: StaticPrice
+    price: 2000
+    #Imperial Space Pirates: New Horizon
```
Комментируйте каждую строчку. Либо закрывайте ваши изменения в комментариях `Imperial Space название_разработки Start/End`

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Weapons/Bombs/pipebomb.yml:
```diff
@@ -1,5 +1,5 @@
 - type: entity
-  parent: GrenadeBase
+  parent: [BaseItem, BaseMajorContraband] #Imperial Space Pirates: New Horizon
```
Создайте отдельную пайп-бомбу и замените обыную на ту, что вы сделаете в файле миграции.

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml:
```diff
@@ -618,7 +618,10 @@
     weight: 0.0002 # 5,000 times less likely than 1 regular animal
     # not putting a BlockMovement component here cause that's funny.
-
+  - type: GPSTrackerPrice
```
Комментарии

krevetkqqq replied:
есть

The-Fiafi commented on Resources/Prototypes/Roles/Antags/pirate.yml:
```diff
@@ -1,25 +0,0 @@
-- type: startingGear
```
Зачем вы это удалили? Используйте другие названия для гиров с какой-либо припиской (ImperialPirateGear и т.д)

The-Fiafi commented on Resources/Prototypes/Roles/Jobs/Fun/visitors_startinggear.yml:
```diff
@@ -1834,12 +1834,11 @@
     id: PiratePDA
     back: ClothingBackpackSatchelLeather
     belt: ClothingBeltUtility
-    ears: ClothingHeadsetFreelance
```
Комментарий

The-Fiafi commented on Resources/Prototypes/Roles/Jobs/Fun/visitors_startinggear.yml:
```diff
@@ -1834,12 +1834,11 @@
   id: PirateScoonerAltA
   equipment:
     jumpsuit: ClothingUniformJumpsuitPirate
-    shoes: ClothingShoesBootsLaceup
```
Комментарий

The-Fiafi commented on Resources/Prototypes/secret_weights.yml:
```diff
@@ -6,8 +6,10 @@
+    Revolutionary: 0.1
     Wizard: 0.05 # Why not, should probably be lower
+    #Pirates: 0.1
+#please remove the comment on private repository
```
Уберите из этого файла все. Тут нет нормальных весов режимов

The-Fiafi commented on Content.Server/Imperial/PiratesNewHorizon/Rules/Systems/PiratesRuleSystem.cs:
```diff
@@ -0,0 +1,211 @@
+using System.Linq;
```
Пожалуйста, удалите нетспользуемые системы.

The-Fiafi commented on Content.Server/Imperial/PiratesNewHorizon/Rules/Systems/PiratesRuleSystem.cs:
```diff
@@ -0,0 +1,211 @@
+                var pirates = ent.Comp;
+                pirates.InitialItems.Clear();
+                pirates.PirateShuttle = GetShuttle((ent, ent)) ?? EntityUid.Invalid;
+                pirates.InitialShipValue = _pricingSystem.AppraiseGrid(shuttle.Owner, uid =>
```
Используйте uid, который вам поставляет EntityQueryEnumerator

The-Fiafi commented on Content.Server/Imperial/PiratesNewHorizon/Rules/Systems/PiratesRuleSystem.cs:
```diff
@@ -0,0 +1,211 @@
+    }
+    protected override void AppendRoundEndText(EntityUid uid, PiratesRuleComponent comp, GameRuleComponent gameRule, ref RoundEndTextAppendEvent ev)
+    {
+        var ent = comp.Owner;
```
Используйте uid

The-Fiafi commented on Content.Server/Imperial/PiratesNewHorizon/Trigger/Systems/TriggerSystemOnHeat.cs:
```diff
@@ -0,0 +1,110 @@
+            toUpdate.Add((uid, trigger));
+        }
+
+        foreach (var a in toUpdate)
```
Зачем вы это вынесли в отдельный цикл? Почему сразу не выполнять этот кусок кода вместо добавления в лист?

The-Fiafi commented on Content.Server/Imperial/PiratesNewHorizon/EntityEffects/ResistChange.cs:
```diff
@@ -0,0 +1,79 @@
+            var offsetTime = Math.Max(status.ModifierTimer.TotalSeconds, gameTiming.CurTime.TotalSeconds);
+
+            status.ModifierTimer = TimeSpan.FromSeconds(offsetTime + time);
+            status.Dirty();
```
Используйте IEntityManager

krevetkqqq replied:
```suggestion
            Dirty(status.Owner, status);
```

The-Fiafi reviewed (commented):
Код, в целом, хороший. Парочка недочетов имеется. Апстримните ветку. На будущее, пожалуйста, посмотрите конвенции по правильному стилю написания (иль используйте линтеры с ИИ)

Maggotson commented:
Санк ю! Как вернусь с Артека запилю

Maggotson committed:
Pirates: New Horizon; Ultrafix (Prepare thyself)

Пофиксил системку резист эффекта (я забыл добавить системку, упс)
Выполнены требования разработчиков по усовершенствованию кода
Добавлены комментарии!!!(Йей!!!)

Maggotson commented:
> Новые или изменённые мапы, гриды создать/скопировать в Imperial/Maps/название разработки. Цены с gps трекером снизить в 1.5/2 раза Добавить медиа контент в раздел ПР`а "Медиа" - видеоролик с обзором Ваших антагонистов и изменений визовского кода

По какой причине снизить цену с gps трекером. Как тогда иначе зарабатывать 300к, если такой обесцен хайрисков?

Maggotson committed:
Merge branch 'develop' into develop

Maggotson committed:
Quick Upstream Fix

Maggotson committed:
paper/folders separation quick fix

Maggotson committed:
omfg the biggest fix in the history of space station 14

Maggotson committed:
This one is biggest fix by far

Maggotson committed:
ROLE TYPE FIX PIRATES

The-Fiafi commented:
Исправьте ошибки YAML линтера

Maggotson committed:
Easy Fix Pirates New Horizon

Maggotson committed:
Упразднение гига пирата

Maggotson commented:
Линтер пофиксил, остальное сломал не я

KotingGG commented:
да тут целая война

Maggotson committed:
Instaquick fix

Maggotson committed:
коммит против вульгарных слов

Maggotson committed:
баланс бафф

Maggotson commented:
Ура size/Medium, а не small 🫶

Maggotson commented:
Дайте label antagonists 😔

Maggotson committed:
quicky tippy fix

Maggotson committed:
pivovarnya228

The-Fiafi commented on Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:
```diff
@@ -211,15 +216,28 @@ private bool CanSell(EntityUid uid, TransformComponent xform)
+        if (!component.SpawnStack)
         {
-            return;
+            if (_station.GetOwningStation(uid) is { } owningStation && 
```
Закомментируйте код который вы убрали в этом файле, иначе все изменения потеряются при апстримах

krevetkqqq replied:
@The-Fiafi закомментируйте?))))) xd 
219 строка

The-Fiafi commented on Resources/Prototypes/Roles/MindRoles/mind_roles.yml:
```diff
@@ -305,5 +305,6 @@
     antagPrototype: Zombie
     exclusiveAntag: true
-    subtype: role-subtype-zombie
+    roleType: TeamAntagonist
```
Зачем?

krevetkqqq replied:
```suggestion
    subtype: role-subtype-zombie
```

The-Fiafi commented on Resources/Prototypes/secret_weights.yml:
```diff
@@ -6,8 +6,7 @@
     Zombie: 0.05
     Zombieteors: 0.05
-    Survival: 0.15
+    Survival: 0.1
```
Уберите все изменения

krevetkqqq replied:
```suggestion
    Survival: 0.15
```

The-Fiafi commented on Resources/Prototypes/ai_factions.yml:
```diff
@@ -10,6 +10,7 @@
   - AllHostile
   - Wizard
   - SyndicateAgent # Imperial Syndicate Agent Fix
+  - Pirate
```
Комментарии

krevetkqqq replied:
```suggestion
  - Pirate # Imperial Space Pirates: New Horizon
```
везде

The-Fiafi commented on Resources/Prototypes/Roles/Jobs/Fun/visitors_startinggear.yml:
```diff
@@ -1871,10 +1872,9 @@
-    ears: ClothingHeadsetAltFreelancer
-    neck: ClothingNeckCloakPirateCap
-    outerClothing: ClothingOuterVestWeb
+    head: ClothingHeadHatPirateCaptain
```
Комментарии

krevetkqqq replied:
```suggestion
    ears: ClothingHeadsetAltFreelancer
    neck: ClothingNeckCloakPirateCap
    outerClothing: ClothingOuterVestWeb
```
?

krevetkqqq replied:
тоже самое

The-Fiafi commented on Resources/Prototypes/Recipes/Crafting/Graphs/improvised/pipebomb.yml:
```diff
@@ -31,7 +31,7 @@
         doAfter: 2
   - node: pipebomb
-    entity: PipeBomb
+    entity: PipeBombNewRemastered
```
Зачем? Вы заменили пайп бомбу в файле миграций

krevetkqqq replied:
```suggestion
    entity: PipeBomb
```
кстати реально

The-Fiafi commented on Resources/Maps/Shuttles/pirate.yml:
```diff
@@ -21,6 +21,7 @@ entities:
   entities:
   - uid: 1
     components:
+    - type: PiratesShuttle
```
Комментарии

krevetkqqq replied:
```suggestion
    - type: PiratesShuttle # Imperial Space Pirates: New Horizon
```

The-Fiafi commented on Resources/Audio/Effects/attributions.yml:
```diff
@@ -241,6 +241,7 @@
   copyright: 'created by Vrymaa on Freesound and modified by Chaboricks'
   license: "CC0-1.0"
   source: https://freesound.org/people/Vrymaa/sounds/785128/
+  source: https://freesound.org/people/Vrymaa/sounds/785128/
```
Зачем?

krevetkqqq replied:
```suggestion
```

### Все новые файлы должны быть по пути .../Imperial/...
Свести к минимуму изменения оригинального кода - это значит, что максимум допускается вызов необходимой функции в оригинальном коде, обозначенный комментарием "Imperial Название разработки"

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Tools/t-ray.yml:
```diff
@@ -25,3 +25,7 @@
       Glass: 150
   - type: StaticPrice
     price: 85
+  #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
  #Imperial Space Pirates: New Horizon start
```

krevetkqqq replied:
```suggestion
  #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Imperial/Other/TFG/Weapon/M2108.yml:
```diff
@@ -47,7 +47,10 @@
   - type: StealTarget
     stealGroup: M2108
-
+  #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
  #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Specific/Medical/handheld_crew_monitor.yml:
```diff
@@ -31,7 +31,10 @@
   - type: StealTarget
     stealGroup: HandheldCrewMonitor
-
+  #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
  #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml:
```diff
@@ -33,6 +33,10 @@
     - HighRiskItem
   - type: StealTarget
     stealGroup: Hypospray
+  #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
  #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -236,6 +244,13 @@
     sprite: Clothing/Eyes/Hud/medpatch.rsi
   - type: Clothing
     sprite: Clothing/Eyes/Hud/medpatch.rsi
+    #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
    #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -252,6 +267,13 @@
     sprite: Clothing/Eyes/Hud/secpatch.rsi
   - type: Clothing
     sprite: Clothing/Eyes/Hud/secpatch.rsi
+    #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
    #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -268,6 +290,14 @@
     sprite: Clothing/Eyes/Hud/beerpatch.rsi
   - type: Clothing
     sprite: Clothing/Eyes/Hud/beerpatch.rsi
+    #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
    #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -284,6 +314,13 @@
     sprite: Clothing/Eyes/Hud/diagpatch.rsi
   - type: Clothing
     sprite: Clothing/Eyes/Hud/diagpatch.rsi
+    #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
    #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Clothing/Eyes/misc.yml:
```diff
@@ -28,6 +28,10 @@
     sprite: Clothing/Eyes/Misc/eyepatch.rsi
   - type: EyeProtection
     protectionTime: 5
+    #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
    #Imperial Space Pirates: New Horizon start
```

The-Fiafi commented on Resources/Prototypes/Entities/Objects/Misc/dat_fukken_disk.yml:
```diff
@@ -26,7 +26,10 @@
   - type: StealTarget
     stealGroup: NukeDisk
-
+    #Imperial Space Pirates: New Horizon
```
#Imperial Space Pirates: New Horizon; Start/End

krevetkqqq replied:
```suggestion
    #Imperial Space Pirates: New Horizon start
```

The-Fiafi reviewed (changes requested):
Комментарии ставятся на этот же строке с изменениями.

krevetkqqq commented on Resources/Prototypes/Roles/MindRoles/mind_roles.yml:
```diff
@@ -305,5 +305,6 @@
-    subtype: role-subtype-zombie
+    roleType: TeamAntagonist
   - type: ZombieRole
+    subtype: role-subtype-zombie
```
```suggestion
```

krevetkqqq commented:
у меня от твоего диффа комп залагал

krevetkqqq commented on Resources/Prototypes/game_presets.yml:
```diff
@@ -39,6 +39,7 @@
     - Traitor
     - Revolutionary
     - Zombie
+    #- Pirates
```
```suggestion
    #- Pirates # Imperial Space Pirates: New Horizon
```

krevetkqqq commented on Resources/Prototypes/secret_weights.yml:
```diff
@@ -6,8 +6,7 @@
-    Revolutionary: 0.15
-    Wizard: 0.05 # Why not, should probably be lower
-# imperial add-on end
+    Revolutionary: 0.1
```
```suggestion
    Revolutionary: 0.15
```

krevetkqqq commented on Resources/migration.yml:
```diff
@@ -608,6 +608,9 @@ PaxChemistryBottle: ChemistryBottlePax
 ToxinChemistryBottle: ChemistryBottleToxin
+
+# 2025-05-27 Imperial Space; Обновление пайп бомбы
+PipeBomb: PipeBombNewRemastered
```
если собрался менять - меняй сразу прототип с комментарием `# Imperial Space Pirates: New Horizon; start end`. лучше в 1 месте, чем в двух

Maggotson committed:
школа каперов

каперы сходили в школу и теперь обучены навыкам. теперь им легче устраивать рдм

Maggotson committed:
Update Resources/Prototypes/secret_weights.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/secret_weights.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/game_presets.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Roles/MindRoles/mind_roles.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Roles/MindRoles/mind_roles.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Audio/Effects/attributions.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/ai_factions.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson commented:
🫶

krevetkqqq commented on Resources/Prototypes/Entities/Objects/Misc/dat_fukken_disk.yml:
```diff
@@ -26,7 +26,10 @@
+    #Imperial Space Pirates: New Horizon
+  - type: GPSTrackerPrice
+    startprice: 0
+    endprice: 13000
```
```suggestion
    endprice: 13000
    #Imperial Space Pirates: New Horizon end
```

krevetkqqq commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -30,6 +30,10 @@
     damageContainers:
     - Inorganic
     - Silicon
+    #Imperial Space Pirates: New Horizon
```
```suggestion
    #Imperial Space Pirates: New Horizon start
```

krevetkqqq commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -85,6 +89,10 @@
+  #Imperial Space Pirates: New Horizon
+  - type: Tag
+    tags:
+    - HudBeer
```
```suggestion
    - HudBeer
    # Imperial Space Pirates: New Horizon end
```

krevetkqqq commented on Resources/Prototypes/Entities/Clothing/Eyes/hud.yml:
```diff
@@ -85,6 +89,10 @@
   - type: StealTarget
     stealGroup: ClothingEyesHudBeer
   - type: SolutionScanner
+  #Imperial Space Pirates: New Horizon
```
```suggestion
  #Imperial Space Pirates: New Horizon start
```

krevetkqqq commented on Resources/Prototypes/Entities/Objects/Devices/pda.yml:
```diff
@@ -1425,7 +1425,8 @@
   - type: Pda
-    id: PirateIDCard
+  #Imperial Space Pirates: New Horizon
+    id: PirateIDCardNewHorizon
```
на той же строчке где изменение сделай коммент. ну или просто миграцию сделай

krevetkqqq commented on Resources/Prototypes/Entities/Objects/Specific/Medical/handheld_crew_monitor.yml:
```diff
@@ -31,7 +31,10 @@
+  #Imperial Space Pirates: New Horizon
+  - type: GPSTrackerPrice
+    startprice: 0
+    endprice: 5500
```
```suggestion
    endprice: 5500
    #Imperial Space Pirates: New Horizon end
```

krevetkqqq commented on Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml:
```diff
@@ -33,6 +33,10 @@
+  #Imperial Space Pirates: New Horizon
+  - type: GPSTrackerPrice
+    startprice: 0
+    endprice: 7000
```
```suggestion
    endprice: 7000
    #Imperial Space Pirates: New Horizon end
```

krevetkqqq commented on Resources/Prototypes/Imperial/Other/TFG/Weapon/M2108.yml:
```diff
@@ -47,7 +47,10 @@
+  #Imperial Space Pirates: New Horizon
+  - type: GPSTrackerPrice
+    startprice: 0
+    endprice: 13000
```
```suggestion
    endprice: 13000
    #Imperial Space Pirates: New Horizon end
```

krevetkqqq commented on Resources/Prototypes/Roles/Jobs/Fun/visitors_startinggear.yml:
```diff
@@ -1834,12 +1834,13 @@
     back: ClothingBackpackSatchelLeather
     belt: ClothingBeltUtility
-    ears: ClothingHeadsetFreelance
+    #Imperial Space Pirates: New Horizon; Line Removed: "ears: ClothingHeadsetFreelance"
```
bruh не удаляй строчки. комментируй/заменяй внутренность и напиши типо "# string # Imperial Space Pirates: New horizon". как с shoes. а тут просто закомментируй вот так
```suggestion
    # ears: ClothingHeadsetFreelance #Imperial Space Pirates: New Horizon
```

krevetkqqq commented on Resources/Textures/Imperial/PiratesNewHorizon/alhocolessence.rsi/meta.json:
```diff
@@ -0,0 +1,31 @@
+        "y": 32
+    },
+    "license": "CC-BY-SA-3.0",
+    "copyright": "By Ham Maggotson",
```
```suggestion
    "copyright": "Made by "hammaggotson" (discord) for Imperal Space server Space Station 14",
```
вообще как таковой политики меты на империале я не видел, но на других серверах выгляди это так

krevetkqqq commented on Resources/Textures/Imperial/PiratesNewHorizon/bureaucracy.rsi/meta.json:
```diff
@@ -0,0 +1,17 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "by ham maggotson (IT IS HORIZONTAL MUNICIPILITY)",
```
```suggestion
  "copyright": "Made by "hammaggotson" (discord) for Imperal Space server Space Station 14", 
```

krevetkqqq commented on Resources/Textures/Imperial/PiratesNewHorizon/clothing/belt/belt_c.rsi/meta.json:
```diff
@@ -0,0 +1,26 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "made by max_gab",
```
а ты у автора спросил под какой лицензией?

Maggotson replied:
он приложил json файлы

krevetkqqq replied:
резолвь тогда

krevetkqqq commented on Resources/Textures/Imperial/PiratesNewHorizon/clothing/belt/gunman.rsi/meta.json:
```diff
@@ -0,0 +1,18 @@
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "made by max_gab",
```
тоже самое

krevetkqqq commented on Resources/Textures/Imperial/PiratesNewHorizon/clothing/belt/redbelt.rsi/meta.json:
```diff
@@ -0,0 +1,18 @@
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "made by max_gab",
```
тоже самое

krevetkqqq commented on Content.Server/Cargo/Components/CargoPalletConsoleComponent.cs:
```diff
@@ -6,4 +6,12 @@ namespace Content.Server.Cargo.Components;
 
 [RegisterComponent]
 [Access(typeof(CargoSystem))]
-public sealed partial class CargoPalletConsoleComponent : Component;
```
этот бро использует устаревший синтаксис. неймспейс деклалировать можно, без тела

krevetkqqq commented on Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:
```diff
@@ -153,8 +155,8 @@ private void GetPalletGoods(EntityUid gridUid, out HashSet<EntityUid> toSell,  o
         toSell = new HashSet<EntityUid>();
-
-        foreach (var (palletUid, _, _) in GetCargoPallets(gridUid, BuySellType.Sell))
+        // Imperial Space: Pirates New Horizon
```
start end

krevetkqqq commented on Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:
```diff
@@ -211,15 +216,28 @@ private bool CanSell(EntityUid uid, TransformComponent xform)
+        EntityUid station = default;
+        StationBankAccountComponent? bankAccount = null;
+        bool hasBankAccount = false;
+        // i hate this crap i had to rewrite it cause i change 1 single line of code (and got 6 fricking errors)
```
м

krevetkqqq commented on Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:
```diff
@@ -8,6 +8,8 @@
 using Robust.Shared.Audio;
 using Robust.Shared.Prototypes;
+//Imperial Space Pirates: New Horizon
+using Content.Shared.Stacks;
```
на той же строчке коммент. я тебе тут не сделаю предложение

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Misc/dat_fukken_disk.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Textures/Imperial/PiratesNewHorizon/bureaucracy.rsi/meta.json

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Textures/Imperial/PiratesNewHorizon/alhocolessence.rsi/meta.json

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Misc/dat_fukken_disk.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/hud.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/misc.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Clothing/Eyes/misc.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Tools/t-ray.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Specific/Medical/handheld_crew_monitor.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Specific/Medical/handheld_crew_monitor.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Roles/Jobs/Fun/visitors_startinggear.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Entities/Objects/Tools/t-ray.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Imperial/Other/TFG/Weapon/M2108.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

Maggotson committed:
Update Resources/Prototypes/Imperial/Other/TFG/Weapon/M2108.yml

Co-authored-by: krevetkqqq <161033382+krevetkqqq@users.noreply.github.com>

The-Fiafi commented:
Исправьте ошибки линтера и валидатора

Maggotson committed:
fix

Maggotson committed:
ghost role fix

Maggotson committed:
Иконки + шмотки + гост роль фикс и короче любя вас

Maggotson committed:
вво страна чудес

krevetkqqq commented:
перезапроси у орикса ревью

krevetkqqq commented on Resources/Prototypes/game_presets.yml:
```diff
@@ -62,6 +62,7 @@
     - Revolutionary
     - Zombie
     - Wizard
+    - Pirates # Imperial Space Pirates: New Horizon
```
оставь просто - Pirates, как я понял

krevetkqqq commented on Resources/Prototypes/Entities/Mobs/Player/ShuttleRoles/settings.yml:
```diff
@@ -1323,7 +1323,7 @@
     - type: NpcFactionMember
       factions:
-      - Syndicate
+      - Pirate
```
комментарии везде

Maggotson replied:
Забыл

krevetkqqq commented on Resources/Textures/Imperial/PiratesNewHorizon/alhocolessence.rsi/meta.json:
```diff
@@ -6,7 +6,7 @@
     },
     "license": "CC-BY-SA-3.0",
-    "copyright": "Made by "hammaggotson" (discord) for Imperal Space server Space Station 14",
+    "copyright": "Made by hammaggotson (discord) for Imperal Space server Space Station 14",
```
чтобы ставить спец символы, можно перед ним поставить обратный слеш. `"Made by \"hammaggotson\" (discord) for Imperal Space server Space Station 14"`

Maggotson committed:
фы

Maggotson committed:
Quick comment

KotingGG committed:
Update gamerules.yml по просьбе HamMaggotson.

Maggotson committed:
Quick balance
